src/intersections/lines.jl

```diff
@@ -41,16 +41,23 @@ end
 
 # compute the intersection of two lines assuming that it is a point
 function intersectpoint(l1::Line, l2::Line)
-  a, b = l1(0), l1(1)
-  c, d = l2(0), l2(1)
+  p1 = first(intersectparameters(l1, l2))
+  l1(0) + p1 * (l1(1) - l1(0))
```

Comment:
Please restructure this code so that the function call is in its own line as

```julia
p1, p2 = intersectparameters(a, b, c, d)
```

and then the final expression becomes:

```julia
a + p1 * (b - a)
```

I think we only need a single function `intersectparameters(a, b, c, d)`. I saw that you added methods with `intersectparameters(l1, l2)` but that is going to become messy with Ray and Segment. We can always resort to the low-level method with 4 points instead.

Reply:
Sure, I'll implement it, good suggestion

Reply:
By the way, looking at segments.jl (intersection of segments)
the variant working with matrix `A`, the rank of augmented / simple matrix also returns the parameters and gives information about whether the lines / segments intersect or not

``` julia
 # get coordinates
  p₁, p₂ = coordinates.(vertices(s1))
  q₁, q₂ = coordinates.(vertices(s2))

  # create matrices and vectors to allow for evaluation of intersection
  A = [(p₂ - p₁) (q₁ - q₂)]
  b = q₁ - p₁

  λ = A \ b
```
I will think about how to use this aspect in the `intersectparameters` function
One idea could be to return the two ranks
``` julia
  # calculate the rank of the augmented matrix
  rₐ = rank([A b])
  # calculate the rank of the rectangular matrix
  r = rank(A)
```

to have 
`p1, p2, r_a, r = intersectparameters(a, b, c, d)`

What do you think?

Reply:
We can discuss these details in the new PR that will be introducing the intersectparameters function. To help future contributors understand our choices.